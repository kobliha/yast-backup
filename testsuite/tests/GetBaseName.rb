# encoding: utf-8

# Backup module - testsuite
#
# GetBaseName tests
#
# testedfiles: backup/functions.ycp
#
# $Id$
#
module Yast
  class GetBaseNameClient < Client
    def main
      Yast.import "UI"
      Yast.import "Testsuite"

      @readmap = {
        "target" => { "string" => "", "size" => -1, "tmpdir" => "/tmp" },
        "proc"   => { "cpuinfo" => { "value" => { "0" => { "flags" => "" } } } },
        "probe"  => {
          "has_smp" => false,
          "is_uml"  => false,
          "cdrom"   => { "manual" => [] }
        }
      }

      @execmap = {
        "target" => {
          "bash_output" => {
            "exit"   => 0,
            "stderr" => "",
            "stdout" => "charmap=\"ISO-8859-1\"\n"
          }
        }
      }

      Testsuite.Init([@readmap, {}, @execmap], nil)

      Yast.include self, "backup/functions.rb"

      Testsuite.Test(lambda { GetBaseName(nil) }, [{}, {}, {}], nil)
      Testsuite.Test(lambda { GetBaseName("") }, [{}, {}, {}], nil)
      Testsuite.Test(lambda { GetBaseName("file") }, [{}, {}, {}], nil)
      Testsuite.Test(lambda { GetBaseName("file.ext") }, [{}, {}, {}], nil)
      Testsuite.Test(lambda { GetBaseName("/") }, [{}, {}, {}], nil)
      Testsuite.Test(lambda { GetBaseName("/file") }, [{}, {}, {}], nil)
      Testsuite.Test(lambda { GetBaseName("/file.ext") }, [{}, {}, {}], nil)
      Testsuite.Test(lambda { GetBaseName("/dir/") }, [{}, {}, {}], nil)
      Testsuite.Test(lambda { GetBaseName("/dir/dir2/file.ext") }, [{}, {}, {}], nil)
      Testsuite.Test(lambda { GetBaseName("/dir.1/dir.2/file") }, [{}, {}, {}], nil)
      Testsuite.Test(lambda { GetBaseName("/dir.1/dir.2/file.ext") }, [
        {},
        {},
        {}
      ], nil)

      nil
    end
  end
end

Yast::GetBaseNameClient.new.main
