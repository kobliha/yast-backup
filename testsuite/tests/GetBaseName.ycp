// Backup module - testsuite
//
// GetBaseName tests
//
// $Id$
//

{
include "testsuite.ycp";

map readmap = $[ "target" : $["string" : "", "size" : -1, "tmpdir" : "/tmp" ],
    "proc" : $["cpuinfo" : $[ "value" : $["0" : $[ "flags" : ""]]]],
    "probe" : $["has_smp" : false, "is_uml" : false, "cdrom" : $["manual" : []]],
    ];

map execmap = $[ "target" : $["bash_output" : $["exit":0, "stderr":"", "stdout":"charmap=\"ISO-8859-1\"\n"]]];

TESTSUITE_INIT([ readmap, $[], execmap ], nil);

include "backup/functions.ycp";

TEST( ``(GetBaseName(nil) ), [ $[], $[], $[] ], nil );
TEST( ``(GetBaseName("") ), [ $[], $[], $[] ], nil );
TEST( ``(GetBaseName("file") ), [ $[], $[], $[] ], nil );
TEST( ``(GetBaseName("file.ext") ), [ $[], $[], $[] ], nil );
TEST( ``(GetBaseName("/") ), [ $[], $[], $[] ], nil );
TEST( ``(GetBaseName("/file") ), [ $[], $[], $[] ], nil );
TEST( ``(GetBaseName("/file.ext") ), [ $[], $[], $[] ], nil );
TEST( ``(GetBaseName("/dir/") ), [ $[], $[], $[] ], nil );
TEST( ``(GetBaseName("/dir/dir2/file.ext") ), [ $[], $[], $[] ], nil );
TEST( ``(GetBaseName("/dir.1/dir.2/file") ), [ $[], $[], $[] ], nil );
TEST( ``(GetBaseName("/dir.1/dir.2/file.ext") ), [ $[], $[], $[] ], nil );

}

