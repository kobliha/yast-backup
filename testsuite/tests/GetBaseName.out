Read	.target.size "/etc/install.inf" -1
Read	.target.tmpdir "/tmp"
Dir	.autoyast2.desktop."s": []
Dir	.yast2.groups.s: []
Read	.probe.architecture nil
Read	.probe.has_apm nil
Read	.probe.has_pcmcia nil
Read	.probe.has_smp false
Read	.probe.system nil
Read	.sysconfig.console.CONSOLE_ENCODING nil
Execute	.target.bash_output "locale -k charmap" $["exit":0, "stderr":"", "stdout":"charmap=\"ISO-8859-1\"\n"]
Read	.target.tmpdir "/tmp"
Read	.probe.floppy.manual nil
Read	.probe.cdrom.manual []
Return	
Return	
Return	file
Return	file
Return	
Return	file
Return	file
Return	
Return	file
Return	file
Return	file
